### DISCUSSION TITLE
Transitionning from Aggregator 2 to 3, a necessity?

### DISCUSSION THREAD

**Discussion by vlaquest** [tfsaggregator/tfsaggregator#363] [Q&A] [answered]
Hi !
We are using Aggregator 2 in server plugin mode, on an Azure DevOps Server 2020.
As you explain in the docs: https://tfsaggregator.github.io/docs/v3/, I understand that I have to move on to v3 because the APIs the server plugin is relying on are obsolete
 => to clarify they announced the object model was obsolete 3 years ago, but it is still working in today's latest version.

Can you please confirm that the APIs the server side plugin uses are really obsolete and will probably soon be discontinuited?
I really liked the "fullly server side" aspect of the TFS Aggregator, there was no roundtrip when performing updates in the "currently updated" work item => it was transparent for the users.

I acknowledge though it is a lot of work for you to compile against every version and manage many binaries, thanks for this.

Is it no longer possible to code server plugins? Or maybe server plugins cannot access work items as directly as before, making a remote approach (azure functions, docker image) almost equivalent?

I also wonder whether the TFS Aggregator could be an Extension, would it then give direct access to work items?

Best regards,
Vincent

**Comment by giuliov** [comment, marked as answer]
Lot of questions.

The Object Model SDK we use in TFS Aggregator are obsolete but still working and Microsoft has no statement when they will stop working. This and the support for Azure DevOps Service was the major driver for the new version which leverages the REST API.
The new Aggregator can be deployed in many ways, the easiest in Azure. On premises there is some manual configuration work and we have some backlog items to simplify it.

The plugin API is not deprecated (though it changed in almost every release of TFS/AzDO), but it is just the notification mechanism: the Rules use the  Object Model SDK.
An Extension does not support notifications, in fact the new Aggregator uses Service Hooks for notification.

Hope this clarifies

**Comment by jessehouwing** [comment]
On top of that: in discussions with the team they clarified that too many
things seem to rely on the old model, but that they aren't testing it
anymore at the same rigidity as before and that new features in Azure
Boards may at some point break cell compatibility or may render some fields
readonly.

With the API being deprecated, it has also stopped changing too much, which
has made it easier to maintain. The last 8 or so point releases required
very little in terms of effort, only the introduction of 2020 and soon e
guess the introduction of Visual Studio 2022 will cause us a lot of new
work. With these major releases all of the assembly strong names change.
And this is the point at which, in the past, many breaking changes were
introduced.

I don't see us not supporting any 2020.* releases, but I'm not terribly
excited to do the work to support any upcoming 2022 version.

**Comment by vlaquest** [comment]
Hi, thanks @giuliov and @jessehouwing for this very detailed insight. This is exactly what I wanted to understand, more about the context, and the current indications given by the product team.
The Aggregator is a beautiful project. I shall upgrade to the docker image in the year to come I think. I wont hesitate to give you some feedback.
Farewell and thanks again for all this work, we can't live without it :)
